### ISSUE TITLE
LSP support for modules

### ISSUE THREAD

**Issue by mhegreberg** [PowerShell/PowerShellEditorServices#1538]
I'd like to get LSP support for modules I use frequently, like MSOnline.

I copied the modules into my bundle_path, but that doesn't seem to import them into the session of powershell that PSES runs

for context, I'm using neovim LSP running in WSL debian, with these settings:

![image](https://user-images.githubusercontent.com/11844475/128398649-aa718089-0e17-4c0b-b453-3f8f5bd40c99.png)
![image](https://user-images.githubusercontent.com/11844475/128398773-fe9f5b2d-6991-4792-b3b6-8d42cdbaac4d.png)

**Comment by andschwa** [comment]
I'm sorry, what are you trying to do? The `bundle_path` is _just_ for the bundled modules that are necessary for PSES to work. Other PowerShell modules should be installed and loaded like [normal](https://docs.microsoft.com/en-us/powershell/scripting/developer/module/installing-a-powershell-module?view=powershell-7.1).

**Comment by mhegreberg** [comment]
I'm hoping to have lsp support for the modules I'm using, so I can have goto definitions, autocomplete, etc.

when I use a python, typescript, or c# LSP any imported modules/packages show up in those places.

are you saying that if I have the module loaded in the instance of powershell that PSES is running in, then those features should work?

**Comment by andschwa** [comment]
Yes, it should. Essentially the LSP server hosts a PowerShell session, and any modules installed and imported into that session (the integrated terminal) then become available for the server to answer gotos, completions, etc.

**Comment by mhegreberg** [comment]
In that case, I'm not entirely sure what's causing my issue.
here are my module paths in powershell:
![image](https://user-images.githubusercontent.com/11844475/132881537-1a5c473e-20ad-4b2f-a0c8-4fc11d474910.png)

and here is the MSOnline module in two of those paths (I made an extra copy while troubleshooting):
![image](https://user-images.githubusercontent.com/11844475/132881683-599dcb62-9324-46f0-aa71-c6c98c420d52.png)
![image](https://user-images.githubusercontent.com/11844475/132881699-689bcdb7-a2a4-4db9-9729-6f347570971b.png)

WSL is definitely talking to the windows version of powershell:
![image](https://user-images.githubusercontent.com/11844475/132882189-996a6056-e2c1-4aa7-9620-9cf049e74714.png)

the LSP works for some functions:
![image](https://user-images.githubusercontent.com/11844475/132882238-da7c6f05-5b3f-40b8-a387-7691602f202f.png)


but it is not able to pull anything from MSOnline:
![image](https://user-images.githubusercontent.com/11844475/132882295-4a81bfe6-c30c-491a-8565-4a2e6571d5e6.png)

it also seems unable to pull anything from ActiveDirectory, which is also installed:
![image](https://user-images.githubusercontent.com/11844475/132882411-88ccb364-8748-4749-bd63-9031a0db0600.png)


Is there something I'm missing?

**Comment by andschwa** [comment]
Have you tried `Import-Module MSOnline` in the integrated console? Also before you do that, show us the output of `Get-Module` (my guess is that MSOnline does not show up, but should after the import).

**Comment by mhegreberg** [comment]
I haven't messed with the integrated terminal yet, though I suspect you're right. what do I need to do to hook into the repl? i'm not using coc-powershell, but the lsp engine built directly into neovim

**Comment by andschwa** [comment]
Ohhh. Ok, skip that, and just put `Import-Module MSOnline` in your script and try executing it with whatever NeoVim integration is available for executing code in the PowerShell session.

**Comment by andschwa** [comment]
I don't see a whole lot in their [documentation](https://github.com/neovim/nvim-lspconfig/blob/master/CONFIG.md#powershell_es)...a really cheap hack would be to put that import statement in your profile because PSES _should_ execute it. You might have to figure out how to send LSP configurations over from NeoVim. I used to do that with Emacs and it wasn't the easiest thing in the world 😣

**Comment by andschwa** [comment]
@mhegreberg Hey, did you figure this out? Did any of those suggestions help out?

**Comment by mhegreberg** [comment]
Hey, sorry, I missed this.

I tried importing the module within Start-EditorServices.ps1
![image](https://user-images.githubusercontent.com/11844475/134397640-896de934-5347-4a26-af48-88ed0448621c.png)
and it didn't make a difference. I'll mess with using my profile today and see if that works.

thanks!

**Comment by andschwa** [comment]
Just checking in, anything work yet?

**Comment by mhegreberg** [comment]
I have been unsuccessful so far, but I have some time set aside tomorrow I'm going to use to see if I can make vim communicate with the powershell REPL, or at least open it in another shell.

**Comment by Lofavreel** [comment]
In your original post you have this configuration:
> ![image](https://user-images.githubusercontent.com/11844475/128398773-fe9f5b2d-6991-4792-b3b6-8d42cdbaac4d.png)
Are you still using the "-NoProfile" switch? Because this does exactly what it says: It prevents loading any profiles.

**Comment by mhegreberg** [comment]
Once I added the import lines to my profile, i removed the -NoProfile flag, to no avail On Oct 2, 2021 05:00, Lofavreel ***@***.***> wrote:
In your original post you have this configuration:


Are you still using the "-NoProfile" switch? Because this does exactly what it says: It prevents loading any profiles.
